Matchismo/Matchismo/Controller/HistoryViewController.h

```diff
@@ -0,0 +1,14 @@
+//
+//  HistoryViewController.h
+//  Matchismo
+//
+//  Created by Otto Olkkonen on 16/06/2022.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface HistoryViewController : UIViewController
+
+@property (nonatomic) NSAttributedString *historyText;
```

Comment:
I used `NSAttributedString` since the VC doesn't need to know anything about different cards and how the card symbols should be coloured, it just displays the text that is given to it. I refactored the `NSAttributedString` properties out of the model classes into the Playing/Set Card VCs so they add the attributes to the correct ranges of the description string and those strings are stored in another `NSAttributedString` (so the attributes are not lost) that is then given to the HistoryVC when it is shown. 
Not sure if the way I handled the `descriptionLabel` and `moveHistory` in the card VCs in the best way or if there would've been a better way to handle how the VCs get the text from the `CardMatchingGame` model and display it and then provide the text to the HistoryVC.

Reply:
Yeah it's a slightly tricky one. Obviously I'm not that familiar with the code, but I'm wondering if there's a slight problem in the design here in the sense that we have the `CardMatchingGame` object which vends a string describing some event (`lastMoveDescription`), that string is passed around and in other parts of the app we're having to do a regex search on it, add attributes and so on.  Might be easier if instead you express the "lastMove" as an actual model object (i.e. with bool/enum fields or whatever to indicate the card type, if it's a match... and so on), pass that around the app and only express it as a string as the point where it's needed for the UI.
Does that make sense?

If that's a big refactor and you want to just carry on with the exercise then don't worry too much making this kind of change, but maybe at least have a think about what could be done here.

Reply:
By the way I've also just noticed some potential bugs here where we have an `NSMutableString` and we're assigning it directly to `NSString` properties/variables (for example in `AbstractCardGameViewController` at line 99).  The danger with doing this is that the `NSString` property is referencing something that underneath is still actually an `NSMutableString` instance, and therefore might be changed at any point.  This can cause problems because in code if you have an `NSString` property, you'll usually assume that it's immutable and won't change. 
```
NSMutableString *myMutableString = [@"abc" mutableCopy];
NSString *myImmutableString = myMutableString; // from this, we might expect myImmutableString to always be "abc"
[myMutableString appendString:@"def"]; // but after this line, myImmutableString will now be changed to "abcdef"
```

The usual way to deal with this is to always use copy when assigning to immutable types (NSString, NSArray, NSDictionary etc).  So in the example above we would have:
`NSString *myImmutableString = [myMutableString copy]; // this will now be property immutable`